Tasks/AzureFunctionAppV1/deploymentProvider/ConsumptionWebAppDeploymentProvider.ts

```diff
@@ -0,0 +1,45 @@
+import { AzureRmWebAppDeploymentProvider } from './AzureRmWebAppDeploymentProvider';
+import tl = require('vsts-task-lib/task');
+import { AzureAppService } from 'azurermdeploycommon/azure-arm-rest/azure-arm-app-service';
+import { AzureAppServiceUtility } from 'azurermdeploycommon/operations/AzureAppServiceUtility';
+
+export class ConsumptionWebAppDeploymentProvider extends AzureRmWebAppDeploymentProvider {
+
+    public async PreDeploymentStep() {
+        this.appService = new AzureAppService(this.taskParams.azureEndpoint, this.taskParams.ResourceGroupName, this.taskParams.WebAppName, 
+            this.taskParams.SlotName, this.taskParams.WebAppKind);
+        this.appServiceUtility = new AzureAppServiceUtility(this.appService);
+    }
+ 
+    public async DeployWebAppStep() {
+        var storageDetails =  await this.findStorageAccount();
+    }
+
+    private async findStorageAccount() {
+        let appSettings = await this.appService.getApplicationSettings();
+        var storageData = {};
+        if(appSettings && appSettings.properties && appSettings.properties.AzureWebJobsStorage) {
+            let str = appSettings.properties.AzureWebJobsStorage;
+            let dictionary = getKeyValuePairs(str);
+            tl.debug(`Storage Account is: ${dictionary["AccountName"]}`);
+            storageData["AccountName"] = dictionary["AccountName"];
+            storageData["AccountKey"] = dictionary["AccountKey"];
+        }
+        else {
```

Comment:
Should we throw exception even when the account data is not present in the app setting. This will ensure early failure.

Reply:
It is done.